lxltools/datacompiler.py

```diff
@@ -228,11 +238,16 @@ def _serialize(data):
 def _read_csv(fpath, encoding='utf-8'):
     csv_dialect = CSV_FORMATS.get(fpath.suffix)
     assert csv_dialect
-    with fpath.open('rb') as fp:
+    if unicode is str:
+        opened = fpath.open('rt', encoding=encoding)
+        decode = lambda v: v
+    else:
+        opened = fpath.open('rb')
+        decode = lambda v: v.decode(encoding)
+    with opened as fp:
         reader = csv.DictReader(fp, dialect=csv_dialect)
         for item in reader:
-            yield {k: v.decode(encoding).strip()
-                            for (k, v) in item.items() if v}
+            yield {k: v.strip() for (k, v) in item.items() if v}
```

Comment:
You've dropped `decode` from this line. Can the variable be removed from above?

Reply:
Good catch! I meant to use the conditionally defined `decode` function, which is used with supplied encoding on Python 2, where the CsvReader yields `bytes` objects. This is now adjusted.